Sources/Purchasing/TrialOrIntroPriceEligibilityChecker.swift

```diff
@@ -267,3 +287,68 @@ extension TrialOrIntroPriceEligibilityChecker {
 // @unchecked because:
 // - Class is not `final` (it's mocked). This implicitly makes subclasses `Sendable` even if they're not thread-safe.
 extension TrialOrIntroPriceEligibilityChecker: @unchecked Sendable {}
+
+// MARK: - Diagnostics
+
+private extension TrialOrIntroPriceEligibilityChecker {
+
+    func trackTrialOrIntroEligibilityRequestIfNeeded(startTime: Date,
+                                                     requestedProductIds: Set<String>,
+                                                     result: [String: IntroEligibility],
+                                                     error: Error?,
+                                                     storeKitVersion: StoreKitVersion) {
+        guard #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *),
+              let diagnosticsTracker = self.diagnosticsTracker else {
+            return
+        }
+
+        var unknownCount, ineligibleCount, eligibleCount, noIntroOfferCount: Int?
+        if !result.isEmpty {
+            (unknownCount, ineligibleCount, eligibleCount, noIntroOfferCount) = result.reduce(into: (0, 0, 0, 0)) {
+                switch $1.value.status {
+                case .unknown:
+                    $0.0 += 1
+                case .ineligible:
+                    $0.1 += 1
+                case .eligible:
+                    $0.2 += 1
+                case .noIntroOfferExists:
+                    $0.3 += 1
+                }
+            }
```

Comment:
Note that this would mean sending `nil` counts only when `result` is empty. Because `requestedProductIds` is never empty at this point, this can only happen in case of error necessarily. 

However, not all errored cases will necessarily return an empty `result`, as there are some fallback checks when certain errors happen (e.g. if using SK1 and eligibility cannot be calculated from the local receipt, then the info is requested from the backend).

This means that there can be situations where `error != nil` but still we've returned the nonempty `result`.